src/lib/mcleece/actions.h

```diff
@@ -16,153 +17,172 @@
 namespace mcleece {
 namespace actions {
 	static const int MAX_MESSAGE_LENGTH = 0x100000;
-	static const unsigned MESSAGE_HEADER_SIZE = mcleece::session_header_size() + crypto_secretbox_macbytes();
 
-	inline int keypair(unsigned char* pubk, unsigned char* secret)
+	template <int MODE>
+	inline int generate_keypair(std::string pubk_path, std::string secret_path, std::string pw)
```

Comment:
The basic breakdown is now:

actions: higher level functions
simple: just the basics
cbox: just the basics, but with an extra libsodium crypto_box step